src/Runner.Sdk/Util/WhichUtil.cs

```diff
@@ -128,5 +128,14 @@ public static string Which(string command, bool require = false, ITraceWriter tr
 
             return null;
         }
+
+        // checks if the file is a symlink and if the symlink`s target exists.
+        private static bool IsPathValid(string path, ITraceWriter trace = null)
+        {
+            var fileInfo = new FileInfo(path);
+            if(fileInfo.LinkTarget == null || File.Exists(fileInfo.LinkTarget)) return true;
```

Comment:
`fileInfo.LinkTarget` can be a relative path. 

The osx tests are failing, because `tar` on osx is a symlink to `bsdtar`: 
`lrwxr-xr-x     1 root   wheel         6 24 Aug 10:59 tar -> bsdtar`

This relative path fails the `File.Exists(fileInfo.LinkTarget))` check





consider allowing `fileInfo.Directory + Path.PathSeparator + fileInfo.LinkTarget`

Reply:
thanks @fhammerl , i will take a look, how can I run this tests that are related to other OS? I was planning to install an osx virtual machine, is there an easier solution?

Reply:
added changes